### PR TITLE
build login screen

### PR DESCRIPTION
**Pull request by zuhair5**
## jira
[TS-7](https://shop-go.atlassian.net/browse/TS-7)

**Description:**
* Built a log-in screen.
* Autolayout has been done on all the ui elements.

**How to test it?**
* Go to info.plist and change the storyboard name to Loginscreen.

ScreenShots&Gifs:
Screen Shot 2022-09-20 at 5 37 15 PM

### CODE REVIEW

File: ShopGo/LoginViewController.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+//
+//  LoginViewController.swift
+//  ShopGo
+//
+//  Created by Quratulain on 9/21/22.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController {
```
Comment (haroonmaqsoodch) [inline ShopGo/LoginViewController.swift:10]: one space after class declaration

File: ShopGo/LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        guard let usernameImage = UIImage(named: "user") else {return}
+        addLeftImageToTextfield(textfield: usernameTextfield, addImage: usernameImage)
```
Comment (haroonmaqsoodch) [inline ShopGo/LoginViewController.swift:14]: make a configure method and put these in that, call that method in viewDidLoad()

File: ShopGo/CustomButton.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+//  CustomButton.swift
+//  ShopGo
+//
+//  Created by Quratulain on 9/21/22.
+//
+
+import UIKit
+
+class CustomButton : UIButton {
+    override init(frame: CGRect) {
```
Comment (basitalich) [inline ShopGo/CustomButton.swift:11]: add space between methods.

File: ShopGo/CustomButton.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        super.init(frame: frame)
+        setRadiusAndShadow()
+    }
+    required init?(coder aDecoder: NSCoder) {
```
Comment (basitalich) [inline ShopGo/CustomButton.swift:15]: space here

File: ShopGo/CustomButton.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        super.init(coder: aDecoder)
+        setRadiusAndShadow()
+    }
+    func setRadiusAndShadow() {
```
Comment (basitalich) [inline ShopGo/CustomButton.swift:19]: space here

File: ShopGo/LoginViewController.swift [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        guard let passwordImage = UIImage(named: "padlock") else {return}
+        addLeftImageToTextfield(textfield: passwordTextfield, addImage: passwordImage)
+    }
+    @IBOutlet weak var scrollView: UIScrollView!
```
Comment (basitalich) [inline ShopGo/LoginViewController.swift:18]: space between methods and outlets.

File: ShopGo/LoginViewController.swift [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    @IBOutlet weak var usernameTextfield: UITextField!
+    @IBOutlet weak var passwordTextfield: UITextField!
+    @IBOutlet weak var loginButton: UIButton!
+    @IBOutlet weak var facebookButton: UIButton!
+    @IBOutlet weak var twitterButton: UIButton!
+    @IBOutlet weak var linkedinButton: UIButton!
+}
+extension LoginViewController {
```
Comment (basitalich) [inline ShopGo/LoginViewController.swift:26]: space

**Review by basitalich — CHANGES_REQUESTED** [review]
Add comments to each method you wrote that what it is doing.

File: ShopGo/CustomButton.swift (hunk 3 above)
Comment (zuhair5) [reply ShopGo/CustomButton.swift:11]: Added Spaces between methods

File: ShopGo/CustomButton.swift (hunk 4 above)
Comment (zuhair5) [reply ShopGo/CustomButton.swift:16]: done

File: ShopGo/CustomButton.swift (hunk 5 above)
Comment (zuhair5) [reply ShopGo/CustomButton.swift:22]: space given

File: ShopGo/LoginViewController.swift (hunk 6 above)
Comment (zuhair5) [reply ShopGo/LoginViewController.swift:11]: space given

File: ShopGo/LoginViewController.swift (hunk 2 above)
Comment (zuhair5) [reply ShopGo/LoginViewController.swift]: resolved

File: ShopGo/LoginViewController.swift [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        guard let usernameImage = UIImage(named: "user") else {return}
+        addLeftImageToTextfield(textfield: usernameTextfield, addImage: usernameImage)
+        guard let passwordImage = UIImage(named: "padlock") else {return}
+        addLeftImageToTextfield(textfield: passwordTextfield, addImage: passwordImage)
+    }
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var usernameTextfield: UITextField!
+    @IBOutlet weak var passwordTextfield: UITextField!
```
[the remark this answers is not in the document]
Comment (zuhair5) [reply ShopGo/LoginViewController.swift:13]: resolved

**Review by haroonmaqsoodch — DISMISSED** [review]

**Review by basitalich — APPROVED** [review]

**Review by basitalich — APPROVED** [review]

**Review by haroonmaqsoodch — APPROVED** [review]

**Pull request merged**